### PR TITLE
fix: remove deprecated `create-plugin` script

### LINKED ISSUE #15105: 🐛 Bug Report: `create-app` default template still uses deprecated `create-plugin` command in `package.json.hbs`
### 📜 Description

Generated projects still have the deprecated `create-plugin` script as well as the new `new` script

### 👍 Expected behavior

New projects should not have a deprecated script.

### 👎 Actual Behavior with Screenshots

Generated npm scripts
image

### 👟 Reproduction steps

Follow the steps to create a new project according to https://backstage.io/docs/getting-started/

Check the `$PROJECT_PATH/package.json` file and see the scripts section.

### 📃 Provide the context for the Bug.

The `create-plugin` cli command was deprecated in favour of the `new` command in #13460. But the command was not removed in the template file.

### 🖥️ Your Environment

Not relevant

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by SayakMukhopadhyay**
Signed-off-by: Sayak Mukhopadhyay <mukhopadhyaysayak@gmail.com>

## Hey, I just made a Pull Request!

Fixes #15105 
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/tricky-toys-teach.md
Comment (djamaile) [inline .changeset/tricky-toys-teach.md:5]: This might actually not be clear enough. A user could interpret that the `create-plugin` is removed from the CLI but it isn't. Maybe just a line that the script command is removed from newly generated Backstage apps should be enough 😄

**Review by djamaile — COMMENTED** [review]
This makes sense to me but maybe we want to make this change when we fully remove the `create-plugin` code 🤔 I'll wait for @freben to see what he thinks.

**Comment by SayakMukhopadhyay** [comment]
Then I can make changes to the documentation in this PR itself. Honestly, it would be easier...

File: .changeset/tricky-toys-teach.md
Comment (SayakMukhopadhyay) [reply .changeset/tricky-toys-teach.md:5]: I will come back to this once the scope of this PR is determined

**Comment by SayakMukhopadhyay** [comment]
Let me know if you folks are ok with making doc changes in this PR. I will proceed based on that.

File: packages/create-app/templates/default-app/package.json.hbs
Comment (SayakMukhopadhyay) [reply packages/create-app/templates/default-app/package.json.hbs:23]: The `new` command is the replacement of `create-` style commands. I have not seen any documentation as such except the cli docs.

**Comment by Rugvip** [comment]
@SayakMukhopadhyay yes! please make the documentation changes as part of this PR

File: packages/create-app/templates/default-app/package.json.hbs
Comment (Rugvip) [reply packages/create-app/templates/default-app/package.json.hbs:23]: Yep, `new` is the new `create-plugin`. It's not documented anywhere yet because we wanted to add it to apps first and then start pointing towards it later. Updating the documentation is what we want to do in this PR.

**Comment by awanlin** [comment]
@SayakMukhopadhyay I'm wondering if maybe some coordination with this might make sense: https://github.com/backstage/backstage/pull/15155. You're updating the command but @kurtaking is updating the docs but with the old command details

**Comment by SayakMukhopadhyay** [comment]
> @SayakMukhopadhyay I'm wondering if maybe some coordination with this might make sense: #15155. You're updating the command but @kurtaking is updating the docs but with the old command details

Yeah, that would make sense. But I believe that the doc changes needs to include removal of the old way of doing things too. @kurtaking what do you think? Do you prefer a way to collaborate on this?

**Comment by kurtaking** [comment]
> > @SayakMukhopadhyay I'm wondering if maybe some coordination with this might make sense: #15155. You're updating the command but @kurtaking is updating the docs but with the old command details
> 
> Yeah, that would make sense. But I believe that the doc changes needs to include removal of the old way of doing things too. @kurtaking what do you think? Do you prefer a way to collaborate on this?

It makes sense for me to hold off until you are completed with your changes. I can then commit what I wanted to add to this PR or follow up on my other PR. How does that sound?

**Comment by awanlin** [comment]
Many thanks @SayakMukhopadhyay and @kurtaking for working on these items, as someone who uses these all the time I take for granted some of the gaps in the documentation 🚀

**Comment by kurtaking** [comment]
Hey @SayakMukhopadhyay, how is this work coming along?

**Comment by SayakMukhopadhyay** [comment]
> Hey @SayakMukhopadhyay, how is this work coming along?

Hands full with some mission critical issues at my work. Will get to this once I have some bandwidth. But I am certainly plan to work on this.

**Comment by SayakMukhopadhyay** [comment]
Now that I am going through the docs with a fine toothed comb, I can see random instances where deprecated commands are mentioned in the docs. Some of them are actually confusing me, like `backend:bundle` which seems to be deprecated but still used in the Docker docs.

Would you folks mind if I remove all such instances of deprecated commands being used in the doc? I will also probably rename this PR to reflect that its mostly a doc cleaning PR.

Also, requesting a review on the last commit I made @Rugvip @freben

**Comment by freben** [comment]
Yeah looking at the recent changes using `new` in docs instead of `create-app` is great. That's exactly what we need!

**Comment by SayakMukhopadhyay** [comment]
> Yeah looking at the recent changes using `new` in docs instead of `create-app` is great. That's exactly what we need!

@freben Just clarifying to ensure we are on the same page, shall I increase the scope of this PR to include documentation changes for all other deprecated or removed cli commands?

**Comment by freben** [comment]
Oh just the create-plugin one for starters I think. One by one would be best, in a separate PR, because I think they have different levels of being ready to be replaced

**Comment by SayakMukhopadhyay** [comment]
Then let me rebase this PR and it should be ready to merge pending review.

**Comment by SayakMukhopadhyay** [comment]
Oh, let me rewrite the changeset

File: docs/plugins/create-a-plugin.md
Comment (freben) [inline docs/plugins/create-a-plugin.md:21]: This seems to be lingering still. Last thing remaining I think?

Comment (SayakMukhopadhyay) [reply docs/plugins/create-a-plugin.md:21]: Unfortunate typo 🤣
The output is shown as an image and I was not sure if I should so the output in my windows system. Hence I kept it as such. What do you think? If a powershell output is fine, I can screenshot that and replace the asset.

Comment (freben) [reply docs/plugins/create-a-plugin.md:21]: Oh... Well OK, we can change the asseT separately and just be happy with fixing the typo for now :)

Comment (SayakMukhopadhyay) [reply docs/plugins/create-a-plugin.md]: @freben I updated the image. Let me know if it looks ok, If not, I will amend the commit and not update the image.

**Review by freben — APPROVED** [review]
Looks awesome!

**Review by Rugvip — APPROVED** [review]
Awesome, thank you! 👍 🎉 

Let's :shipit:

**Pull request merged**
